### PR TITLE
[TopBar] Consolidate se23 styles and logic

### LINKED ISSUE #9975: [TopBar] Consolidate se23 logic and styles

### PR DESCRIPTION
**Pull request by laurkim**
### WHY are these changes introduced?

Resolves #9975.

### WHAT is this pull request doing?

Consolidates se23 beta styles and logic for `TopBar`.

### How to 🎩
[Storybook](https://5d559397bae39100201eedc1-hkusscolus.chromatic.com/?path=/story/all-components-topbar--default)
[Prod Storybook](https://storybook.polaris.shopify.com/?path=/story/all-components-topbar--default&globals=polarisSummerEditions2023:true)

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/TopBar/TopBar.scss [hunk 1]
```diff
@@ -106,25 +92,18 @@ $right-column: 1fr;
...
   transition: var(--p-motion-duration-150) fill var(--p-motion-ease)
     var(--p-motion-duration-50);
 
-  #{$se23} & {
-    fill: var(--p-color-icon-inverse);
-  }
-
   &.focused:active {
-    background-color: var(--p-color-bg-active);
+    background-color: var(--p-color-bg-inverse-hover);
```
Comment (laurkim) [inline polaris-react/src/components/TopBar/TopBar.scss:100]: Changed due to specificity issues.

File: polaris-react/src/components/TopBar/components/Menu/Menu.scss [hunk 2]
```diff
@@ -21,106 +21,66 @@ $activator-variables: (
...
-      background-color: var(--p-color-bg-inverse);
-    }
+    margin-right: var(--p-space-2);
+    background-color: var(--p-color-bg-inverse);
 
     &:focus,
     &:hover,
     &:active,
     &[aria-expanded='true'] {
-      background-color: transparent;
```
Comment (laurkim) [inline polaris-react/src/components/TopBar/components/Menu/Menu.scss:123]: Removed due to specificity issues.

File: polaris-react/src/components/TopBar/components/SearchField/SearchField.scss [hunk 3]
```diff
@@ -24,26 +20,19 @@ $search-icon-width-se23: calc(#{$icon-size-se23} + var(--p-space-3));
...
 }
+
 // We have both a focused class and a focus pseudo selector here
 // because we allow "faked" focus for when the search is still
 // active, but is not actually the focused element in the DOM
 // (for example, while selecting a value from a filter in the
 // search this input controls)
 .focused .Input,
 .Input:focus {
-  border: none;
```
Comment (laurkim) [inline polaris-react/src/components/TopBar/components/SearchField/SearchField.scss:34]: Removed `border: none` due to specificity issues.

**Review by sophschneider — APPROVED** [review]

**Review by gwyneplaine — APPROVED** [review]
🚀

**Pull request merged**